q2_types/feature_data/_format.py

```diff
@@ -90,38 +90,59 @@ class TSVTaxonomyFormat(model.TextFileFormat):
 
     Optionally followed by other arbitrary columns.
 
-    This format supports comment lines starting with #, and blank lines. The
-    expected header must be the first non-comment, non-blank line. In addition
-    to the header, there must be at least one line of data.
+    This format supports blank lines. The expected header must be the first
+    non-blank line. In addition to the header, there must be at least one line
+    of data.
 
     """
     HEADER = ['Feature ID', 'Taxon']
 
-    def sniff(self):
+    def _check_n_records(self, n=None):
         with self.open() as fh:
             data_lines = 0
             header = None
-            while data_lines < 10:
-                line = fh.readline()
 
+            file_ = enumerate(fh) if n is None else zip(range(n), fh)
+
+            for i, line in file_:
+                # Tracks line number for error reporting
+                i = i + 1
                 if line == '':
                     # EOF
                     break
                 elif line.lstrip(' ') == '\n':
                     # Blank line
                     continue
 
-                cells = line.rstrip('\n').split('\t')
+                cells = line.strip('\n').split('\t')
+
                 if header is None:
                     if cells[:2] != self.HEADER:
-                        return False
+                        raise ValidationError("['Feature ID' and 'Taxon'] "
+                                              "must be the first two header "
+                                              "values to be valid.\n\n The "
+                                              "first two header values "
+                                              "provided are: {}.\nIssue on "
+                                              "line {}"
+                                              .format(cells[:2], i))
```

Comment:
`cells[:2]` might only be one (or zero) elements long --- please update the error message to account for that (right now it is locked in on "first two header values" which might be confusing if the error shows one (or zero) elements).

```bash
There was a problem importing /home/matthew/Desktop/foo.txt:

  /home/matthew/Desktop/foo.txt is not a(n) TSVTaxonomyFormat file:

  ['Feature ID' and 'Taxon'] must be the first two header values to be valid.

   The first two header values provided are: ['foo'].
  Issue on line 1
```

:kaaba: 

How about:

Header row on line {} is invalid. ['Feature ID' and 'Taxon'] must be the first two header values to be valid. The first two header values provided are: {}.

Reply:
Should ['Feature ID' and 'Taxon'] be left as is? Or changed to [Feature ID, Taxon] which reflects what the user will see when the error message is returned?

Reply:
I don't quite follow the question, can you restate?

Reply:
I added a snippet in my previous comment to provide some context. My thought was to change `['Feature ID' and 'Taxon']` to `['Feature ID', 'Taxon']` in order to appear similar to the raised `Validation Error` for incorrect formatting.

 Please see the revised comment above.

Reply:
Ah thanks for clarifying - I get it ! Yeah, let's go with `['Feature ID', 'Taxon']`